Add content/en/container-orchestration.md

Asmit2952 opened the pull request:
Fixes #327 

Signed-off-by: Asmit Malakannawar <81866614+Asmit2952@users.noreply.github.com>

jihoon-seo commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,14 @@
+---
+
+## What it is
+Container orchestration automates container deployment, management, scaling, and networking. It's primary goal is to manage the life cycle of containers and their dynamic environments.A conductor, like in an orchestra, directs a group of musicians who work collaboratively to perform a piece of music. Similarly containers in your application are orchestra musicians.
```
A following space after the period is missing.

```suggestion
Container orchestration automates container deployment, management, scaling, and networking. It's primary goal is to manage the life cycle of containers and their dynamic environments. A conductor, like in an orchestra, directs a group of musicians who work collaboratively to perform a piece of music. Similarly containers in your application are orchestra musicians.
```

jihoon-seo commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,14 @@
+Container orchestration automates container deployment, management, scaling, and networking. It's primary goal is to manage the life cycle of containers and their dynamic environments.A conductor, like in an orchestra, directs a group of musicians who work collaboratively to perform a piece of music. Similarly containers in your application are orchestra musicians.
+
+## Problem it addresses 
+On behalf of the user, container orchestration organises the compute, networking, and storage infrastructure. To ensure that applications are highly available and computing resources are used to their full potential, it does away with the requirement for direct manual orchestration in a cluster and automates the orchestration process.  
```
I think that currently most words in the CNCF Glossary are in American style.
And I checked [Style Guide](https://glossary.cncf.io/style-guide/), but cannot find sentence that mentions about spelling style (maybe we need this now or later).

Dear maintainers (@CathPag @JasonMorgan @iamNoah1 @seokho-son),
what do you think about this? 😊

```suggestion
On behalf of the user, container orchestration organizes the compute, networking, and storage infrastructure. To ensure that applications are highly available and computing resources are used to their full potential, it does away with the requirement for direct manual orchestration in a cluster and automates the orchestration process.  
```

jihoon-seo commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,14 @@
+On behalf of the user, container orchestration organises the compute, networking, and storage infrastructure. To ensure that applications are highly available and computing resources are used to their full potential, it does away with the requirement for direct manual orchestration in a cluster and automates the orchestration process.  
+
+## How it helps
+Container orchestration results in a more seamless customer experience by increasing productivity, resilience to failure, scalability, and flexibility. In addition to offering a platform for running, managing, and operating containers, it controls the container life cycle. For all container workloads and infrastructure, it offers a centralised management interface that also aids in logging and mo ...
```
```suggestion
Container orchestration results in a more seamless customer experience by increasing productivity, resilience to failure, scalability, and flexibility. In addition to offering a platform for running, managing, and operating containers, it controls the container life cycle. For all container workloads and infrastructure, it offers a centralized management interface that also aids in logging and monitoring.
```

jihoon-seo reviewed (commented):
Thanks for your contribution @Asmit2952.

Some feedback:

- Please use semantic line breaks.
- Please insert empty lines before and after the header lines.

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,14 @@
+On behalf of the user, container orchestration organizes the compute, networking, and storage infrastructure. To ensure that applications are highly available and computing resources are used to their full potential, it does away with the requirement for direct manual orchestration in a cluster and automates the orchestration process.  
+
+## How it helps
+Container orchestration results in a more seamless customer experience by increasing productivity, resilience to failure, scalability, and flexibility. In addition to offering a platform for running, managing, and operating containers, it controls the container life cycle. For all container workloads and infrastructure, it offers a centralized management interface that also aids in logging and mo ...
```
This needs some work too. I (or someone else on our team) will provide more concrete feedback in the next few days.

Asmit2952 replied:
Great, looking forward to it.

CathPag reviewed (commented):
Not ready with my feedback but here's what I got so far...

Asmit2952 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,14 @@
+Container orchestration automates container deployment, management, scaling, and networking. It's primary goal is to manage the life cycle of containers and their dynamic environments. A conductor, like in an orchestra, directs a group of musicians who work collaboratively to perform a piece of music. Similarly containers in your application are orchestra musicians.
+
+## Problem it addresses 
+On behalf of the user, container orchestration organizes the compute, networking, and storage infrastructure. To ensure that applications are highly available and computing resources are used to their full potential, it does away with the requirement for direct manual orchestration in a cluster and automates the orchestration process.  
```
Cool, I'll modify that section.

Asmit2952 replied:
Modified this section, let me know what you think of that

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,14 @@
+On behalf of the user, container orchestration organizes the compute, networking, and storage infrastructure. To ensure that applications are highly available and computing resources are used to their full potential, it does away with the requirement for direct manual orchestration in a cluster and automates the orchestration process.  
+
+## How it helps
+Container orchestration results in a more seamless customer experience by increasing productivity, resilience to failure, scalability, and flexibility. In addition to offering a platform for running, managing, and operating containers, it controls the container life cycle. For all container workloads and infrastructure, it offers a centralized management interface that also aids in logging and mo ...
```
Don't think the customer experience plays a role here (at least not directly), so I'd skip that. Increase productivity, resilience to failure, scalability, and flexibility is the result but not the how. 

My suggestion is above but I'll need Noah's help. Feel free to commit it now, so we can iterate from there.

@iamNoah1, once this is live, we'll get rid of the Kubernetes definition as we don't want to have any definition for particular technologies. I "borrowed" some of the [Kubernetes](https://glossary.cncf.io/kubernetes/)' definitions (which is far too technical btw!). Here are my discussion points:

Not all container orchestrators work declaratively (I think), but since Kubernetes is the clear leader and we won't have a k8s definition soon, we should say that that's how they often work (same with self-healing). Anything else we need to say here?

CathPag reviewed (commented):
Ok, made one more suggestion and started a discussion.

iamNoah1 commented:
@Asmit2952 thanks for the contribution. I will need you to have a little more patience :) It is on my list

iamNoah1 reviewed (changes requested):
@Asmit2952 thanks for your contribution. Before I review in detail, I would ask you to please use semantic line breaks. Also, please try to modify the Problem it addresses section according to the comment from Catherine.

Asmit2952 commented:
> @Asmit2952 thanks for your contribution. Before I review in detail, I would ask you to please use semantic line breaks. Also, please try to modify the Problem it addresses section according to the comment from Catherine.



> @Asmit2952 thanks for your contribution. Before I review in detail, I would ask you to please use semantic line breaks. Also, please try to modify the Problem it addresses section according to the comment from Catherine.

Used semantic line breaks and tried to modify problem it addresses section. Please have a look at your convenience

iamNoah1 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,18 @@
+There were tools that made it simple to run containers within virtual machines, but communicating all of this into the outside world and sharing secrets remained difficult. 
+A platform was required to run, manage, operate, and support the containers throughout their lifecycle.
+
+## How it helps
```
If it is declarative or imperative does not really matter regarding how an orchestrator helps. I would more likely focus on the fact that the orchestrator takes care that all containerized apps are in a certain state. The fact that the orchestrator itself is self healing, does not necessarily ensure that the cluster remains in the desired state. Also cluster is a term that you used the first time in relation to container orchestration. This might confuse readers, and I am not entirely if all container orchestrators are clusters.

Asmit2952 replied:
Okay, will modify according to your suggestions

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Tags:** Brittle

iamNoah1 reviewed (changes requested):
@Asmit2952 super sorry for the delay, but this term is very important and also not easy. I made some suggestions and would like to ask for a rework of the last section (how it helps), based on the direction I gave, if it makes sense to you as well.

CathPag commented:
@Asmit2952, are you still interested in working on this or should we take it from here?

Asmit2952 commented:
Hey @CathPag I am still interested to work on this. I'm finding it a bit difficult to find the relevant content.

CathPag commented:
@Asmit2952, you can also leave it as is for someone else to pick up.

Asmit2952 commented:
> @Asmit2952, you can also leave it as is for someone else to pick up. That way, you helped move it forward but hand it off. 

Hey @CathPag please just give me few more days. I want to work on this. Will update this PR soon.

Asmit2952 commented:
Updated the "how it helps" part

Asmit2952 commented:
Any updates @iamNoah1?

iamNoah1 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,17 @@
+---
+
+## What it is
+[Container](/container/) orchestration provides the ability to manage and automate the lifecycle of containerized applications in dynamic environments. This includes different aspects that are also pursued by [Cloud Native Technologies](/cloud-native-tech/) and [Cloud Native Apps](/cloud-native-apps/), such as: deployment, (auto)scaling, auto healing and monitoring.
```
```suggestion
[Container](/container/) orchestration provides the ability to manage 
and automate the lifecycle of containerized applications in dynamic environments. 
This includes different aspects that are also pursued by [Cloud Native Technologies](/cloud-native-tech/) and [Cloud Native Apps](/cloud-native-apps/), 
such as: deployment, (auto)scaling, auto healing and monitoring.
```

iamNoah1 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,17 @@
+
+## What it is
+[Container](/container/) orchestration provides the ability to manage and automate the lifecycle of containerized applications in dynamic environments. This includes different aspects that are also pursued by [Cloud Native Technologies](/cloud-native-tech/) and [Cloud Native Apps](/cloud-native-apps/), such as: deployment, (auto)scaling, auto healing and monitoring.
+It functions like a data center operating system (OS), managing [containerized](/containerization/) applications at scale (just like the OS on your laptop that manages your apps).
```
I am not sure if this metaphor is accurate. I like the orchestrator of an orchestra metaphor, but that is just my opinion. 

Maybe we can say something like: "It works like a conductor of an orchestra, who makes sure that every part of the orchestra does what it should."

iamNoah1 replied:
```suggestion
It functions like a data center operating system (OS), 
managing [containerized](/containerization/) applications at scale 
(just like the OS on your laptop that manages your apps).
```

iamNoah1 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,17 @@
+It functions like a data center operating system (OS), managing [containerized](/containerization/) applications at scale (just like the OS on your laptop that manages your apps).
+
+## Problem it addresses 
+The rise of [Cloud Native Apps](/cloud-native-apps/) and [technologies](/cloud-native-tech/) as well as the fast changing demands of modern businesses introduced a new set of problems. Managing microservices, security and network related things at scale as well as distributed system in general are hard if not impossible to manage manually.   
```
```suggestion
The rise of [Cloud Native Apps](/cloud-native-apps/) and [technologies](/cloud-native-tech/) 
as well as the fast-changing demands of modern businesses introduced a new set of problems. Managing [microservices](/microservices), security, and network-related things at scale, 
as well as [distributed systems](/distributed-systems) in general, 
are hard if not impossible to manage manually.   
```

iamNoah1 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,17 @@
+## How it helps
+Container orchestration, provides a declarative method of automating much of the work, which makes the operational complexity manageable for development and operations, or DevOps. 
+This includes a variety of tasks that software teams must perform in order to manage the lifecycle of a container, such as provisioning, deployment, scaling (both up and down), networking, load balancing, and other activities.
+Orchestrators are also often [self-healing](https://glossary.cncf.io/self-healing/), ensuring the cluster's actual state always matches the operator's desired state.
```
```suggestion
```

This sentence does not really add to the understanding of what a container orchestrator is.

iamNoah1 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,17 @@
+The rise of [Cloud Native Apps](/cloud-native-apps/) and [technologies](/cloud-native-tech/) as well as the fast changing demands of modern businesses introduced a new set of problems. Managing microservices, security and network related things at scale as well as distributed system in general are hard if not impossible to manage manually.   
+
+## How it helps
+Container orchestration, provides a declarative method of automating much of the work, which makes the operational complexity manageable for development and operations, or DevOps. 
```
```suggestion
Container orchestration provides a declarative method of automating much of the work, 
which makes the operational complexity manageable for software engineering teams.
```

iamNoah1 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,17 @@
+
+## How it helps
+Container orchestration, provides a declarative method of automating much of the work, which makes the operational complexity manageable for development and operations, or DevOps. 
+This includes a variety of tasks that software teams must perform in order to manage the lifecycle of a container, such as provisioning, deployment, scaling (both up and down), networking, load balancing, and other activities.
```
```suggestion
This includes a variety of tasks, such as provisioning, deployment, scaling (both up and down), networking, load balancing, and other activities.
```

iamNoah1 reviewed (changes requested):
@Asmit2952 sorry for the delay, I added some suggestions. Most important are the semantic line breaks. I asked like 3 times to please add those. Please take a look, so that you get a feeling of what they look like. 
However, I simplified some content, please take a look as well and apply or decline the suggestions accordingly. I think we will need some editing from @CathPag afterward ;).

Asmit2952 commented:
> @Asmit2952 sorry for the delay, I added some suggestions. Most important are the semantic line breaks. I asked like 3 times to please add those. Please take a look, so that you get a feeling of what they look like. 
> However, I simplified some content, please take a look as well and apply or decline the suggestions accordingly. I think we will need some editing from @CathPag afterward ;). 
> 
> 

Hey @iamNoah1 I did add semantic line breaks in one of the commit. I added a break after every full stop. Didn't know that I had to add breaks after conjunctions as well.

CathPag commented:
@Asmit2952, are you still interested in working on this, or should we open it up for someone else?

Asmit2952 commented:
Hey @CathPag, sorry for the delay. I missed the notification. I signed-off and committed most of the suggestions. Is there anything left to change? Can you go through the final doc once?

iamNoah1 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,25 @@
+and automate the lifecycle of containerized applications in dynamic environments. 
+This includes different aspects that are also pursued by [Cloud Native Technologies](/cloud-native-tech/) and [Cloud Native Apps](/cloud-native-apps/), 
+such as: deployment, (auto)scaling, auto healing and monitoring.
+It functions like a data center operating system (OS), 
```
```suggestion
It works like a conductor of an orchestra, who makes sure that every part of the orchestra does what it should.
```

iamNoah1 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,25 @@
+This includes different aspects that are also pursued by [Cloud Native Technologies](/cloud-native-tech/) and [Cloud Native Apps](/cloud-native-apps/), 
+such as: deployment, (auto)scaling, auto healing and monitoring.
+It functions like a data center operating system (OS), 
+managing [containerized](/containerization/) applications at scale 
```
```suggestion

```

iamNoah1 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,25 @@
+such as: deployment, (auto)scaling, auto healing and monitoring.
+It functions like a data center operating system (OS), 
+managing [containerized](/containerization/) applications at scale 
+(just like the OS on your laptop that manages your apps).
```
```suggestion

```

iamNoah1 reviewed (changes requested):
@Asmit2952 thanks for everything, though you missed my last comment about the metaphor. I added some suggestion. @CathPag can you have a look if you like it?!. Other than that, I would say it is ready for editing if needed.

iamNoah1 commented:
@Asmit2952 Also apparently the commits are not signed off correctly. Following the instructions of the DCO check, you should be able to fix that: https://github.com/cncf/glossary/pull/993/checks?check_run_id=9440776514.

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,18 @@
+---
+
+## What it is
+[Container](/container/) orchestration provides the ability to manage and automate the lifecycle of containerized applications in dynamic environments. 
```
```suggestion
[Container](/container/) orchestration refers to managing and automating containerized applications' lifecycles in dynamic environments. 
```

Provided the ability to manage = too many words

CathPag replied:
We should probably add in the first section that this is infrastructure specific.

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,18 @@
+
+## What it is
+[Container](/container/) orchestration provides the ability to manage and automate the lifecycle of containerized applications in dynamic environments. 
+This includes different aspects that are also pursued by [Cloud Native Technologies](/cloud-native-tech/) and [Cloud Native Apps](/cloud-native-apps/), such as: deployment, (auto)scaling, auto healing and monitoring.
```
```suggestion
It's executed through a [cloud native technology](/cloud-native-tech/) (in most cases, [Kubernetes[(/kubernetes)) that enables deployments, (auto)scaling, auto-healing, and monitoring. 
Orchestration is a metaphor. 
```
Orchestration tools are cloud native tech

CathPag replied:
```suggestion
It's [cloud native technology](/cloud-native-tech/) that enables deployment, (auto)scaling, auto-healing, and monitoring. 
Orchestration is a metaphor. 
```

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,18 @@
+## What it is
+[Container](/container/) orchestration provides the ability to manage and automate the lifecycle of containerized applications in dynamic environments. 
+This includes different aspects that are also pursued by [Cloud Native Technologies](/cloud-native-tech/) and [Cloud Native Apps](/cloud-native-apps/), such as: deployment, (auto)scaling, auto healing and monitoring.
+It works like a conductor of an orchestra, who makes sure that every part of the orchestra does what it should.
```
```suggestion
The orchestration tool conducts containers like a music conductor, ensuring every container (or musician) does what it should. 
```

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,18 @@
+It works like a conductor of an orchestra, who makes sure that every part of the orchestra does what it should.
+
+## Problem it addresses 
+The rise of [Cloud Native Apps](/cloud-native-apps/) and [technologies](/cloud-native-tech/) as well as the fast-changing demands of modern businesses introduced a new set of problems. 
```
```suggestion
The rise of [cloud native apps](/cloud-native-apps/) and technologies and the fast-changing demands of modern businesses introduced a new set of problems. 
```

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,18 @@
+
+## Problem it addresses 
+The rise of [Cloud Native Apps](/cloud-native-apps/) and [technologies](/cloud-native-tech/) as well as the fast-changing demands of modern businesses introduced a new set of problems. 
+Managing [microservices](/microservices), security, and network-related things at scale, as well as [distributed systems](/distributed-systems) in general, are hard if not impossible to manage manually.   
```
```suggestion
Managing [microservices](/microservices), security, and network communication at scale — [distributed systems](/distributed-systems) in general — are hard, if not impossible, to manage manually.
Container orchestration automates all these management tasks. 
```

network-related things doesn't sound good. I assume it's communication?

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,18 @@
+Managing [microservices](/microservices), security, and network-related things at scale, as well as [distributed systems](/distributed-systems) in general, are hard if not impossible to manage manually.   
+
+## How it helps
+Container orchestration provides a declarative method of automating much of the work, which makes the operational complexity manageable for software engineering teams.
```
```suggestion
Container orchestration tools allow users to determine a system's state declaratively. 
First, they determine (or declare) how it should look like (e.g., x containers, y pods, etc.).
The orchestration tool will then automatically monitor the infrastructure and correct it if its state deviates from the declared one (e.g., spin up a new container if one crashes). 
```

iamNoah1 replied:
Let's modify it a little bit: 
```suggestion
Container orchestration tools allow users to determine a system's state. 
First, they declare how it should look like (e.g., x containers, y pods, etc.).
The orchestration tool will then automatically monitor the infrastructure and correct it if its state deviates from the declared one (e.g., spin up a new container if one crashes). 
```

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,18 @@
+
+## How it helps
+Container orchestration provides a declarative method of automating much of the work, which makes the operational complexity manageable for software engineering teams.
+This includes a variety of tasks, such as provisioning, deployment, scaling (both up and down), networking, load balancing, and other activities.
```
```suggestion
This automation simplifies many of the engineering teams' otherwise highly manual and complex operational tasks, including provisioning, deployment, scaling (up and down), networking, load balancing, and other activities.
```

CathPag reviewed (commented):
@iamNoah1, let me know what you think

iamNoah1 commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,22 @@
+---
+title: Container Orchestration
+status: Draft
```
I like the idea of calling it draft instead of feedback appreciated, but if we introduce that here, we have to consequently change it in all other places, especially in the localization readmes. What about doing that in separate issue/PR?!

CathPag replied:
ok, let's talk about this in our next maintainer meeting. I think this is almost ready, so we can ignore here and update to completed once done.

iamNoah1 replied:
agreed

iamNoah1 reviewed (changes requested):
@Asmit2952 could you please add "lifecycles" to the wordlist. Thanks

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,22 @@
+
+## What it is
+[Container](/container/) orchestration refers to managing and automating the lifecycle of containerized applications in dynamic environments. 
+It's executed through a [cloud native technology](/cloud-native-tech/) (in most cases, [Kubernetes](/kubernetes)) that enables deployments, (auto)scaling, auto-healing, and monitoring. 
```
```suggestion
It's executed through a container orchestrator (in most cases, [Kubernetes](/kubernetes)), which enables deployments, (auto)scaling, auto-healing, and monitoring. 
```

CathPag replied:
It felt like we are forcing cloud native technologies into that sentence

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,22 @@
+## Problem it addresses 
+The rise of [cloud native apps](/cloud-native-apps/) and technologies and the fast-changing demands of modern businesses introduced a new set of problems. 
+Managing [microservices](/microservices), security, and network communication at scale — [distributed systems](/distributed-systems) in general — are hard, if not impossible, to manage manually.
+Container orchestration automates all these management tasks. 
```
```suggestion
Container orchestration allows users to automate all these management tasks. 
```

CathPag commented on content/en/container-orchestration.md:
```diff
@@ -0,0 +1,22 @@
+The orchestration tool conducts containers like a music conductor, ensuring every container (or musician) does what it should. 
+
+## Problem it addresses 
+The rise of [cloud native apps](/cloud-native-apps/) and technologies and the fast-changing demands of modern businesses introduced a new set of problems. 
```
```suggestion

```
This sentence has been bothering me. It doesn't really flow with the rest. Thinking about it today, I don't think we need it. It's a little fluffy and the other two are to the point. Ok if we delete it?

CathPag reviewed (commented):
A few more edits. What do you think, @iamNoah1?

Asmit2952 commented:
> @Asmit2952 Also apparently the commits are not signed off correctly. Following the instructions of the DCO check, you should be able to fix that: https://github.com/cncf/glossary/pull/993/checks?check_run_id=9440776514.

Hey @iamNoah1 , the DCO says the commits should be signed off locally. But I have made changes in web browser itself. Also I read a line "You are absolutely certain nobody else is doing any work based upon this branch", so will these affect yours and Catherine's commits?

Asmit2952 commented:
> @Asmit2952 could you please add "lifecycles" to the wordlist. Thanks

Done!

iamNoah1 reviewed (approved):
@Asmit2952 thanks for the patience and the last edits. Forget about the DCO, I set them to pass so that we don't loose time here :) In the future please make sure to sign your commits locally from the beginning of a contribution. Thank you. @CathPag I guess where are good to go :)

CathPag reviewed (approved):
LGTM
